app/controllers/chargeback_controller.rb

```diff
@@ -208,28 +228,48 @@ def cb_rate_edit
       if params[:typ] == "copy" # if tab was not changed
         session[:changed] = true
         @sb[:rate_details] = []
+        @sb[:num_tiers] = []
+        @sb[:tiers] = []
         rate = ChargebackRate.find(obj[0])
         @sb[:rate] = ChargebackRate.new
         @sb[:rate].description = "Copy of " + rate.description
         @sb[:rate].rate_type = rate.rate_type
-        rate_details = rate.chargeback_rate_details
+        rate_details = rate.chargeback_rate_details.to_a
+        rate_details.sort_by! { |rd| [rd.group.downcase, rd.description.downcase] }
         # Create new rate detail records for copied rate record
-        rate_details.each do |r|
+        rate_details.each_with_index do |r, i|
+          @sb[:tiers][i] = []
           detail = ChargebackRateDetail.new
           detail.description = r[:description]
           detail.source = r[:source]
-          detail.rate = r[:rate]
           detail.per_time = r[:per_time]
           detail.group = r[:group]
           detail.per_unit = r[:per_unit]
           detail.metric = r[:metric]
           detail.chargeback_rate_detail_measure_id = r[:chargeback_rate_detail_measure_id]
+          r.chargeback_tiers.each do |t|
+            tier = ChargebackTier.new
+            tier.start = t[:start]
+            tier.end = t[:end]
+            tier.fix_rate = t[:fix_rate]
+            tier.var_rate = t[:var_rate]
```

Comment:
These assignments are of no concern to the controller, or at least length enough that I suggest you add functionality to the model (or, if complex, a helper class) to be able to pass it attributes that it can then handle itself. This would make the actual logic and loops that you're doing here much easier to see and reason about.